### PR TITLE
Created Auto-Scaling lab #31

### PR DESCRIPTION
**Pull request by Templum**
Hello, 
Created the content for the lab 8 which explains the auto-scaling. The content was created based on the points raised in #31. Please let me know if something is missing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,4 +109,4 @@ The [appendix](./appendix.md) contains some additional content.
 
 ## Acknowledgements
 
-Thanks to @iyovcheva, @BurtonR, @johnmccabe, @laurentgrangeau, @stefanprodan  for testing and contributing to the labs.
+Thanks to @iyovcheva, @BurtonR, @johnmccabe, @laurentgrangeau, @stefanprodan & @templum  for testing and contributing to the labs.
```
Comment (alexellis) [inline README.md]: @kenfdev let's add you to this list too?

File: lab8.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+## Exploring Auto-Scaling:
+
+Before dive into a hands-on session, we will have a brief look how we can configure the auto-scaling out of the box. 
+
+Out of the box Open FaaS is set up in a way that it will auto-scale based on the received `request per seconds`. If the defined threshold for the `request per seconds` is exceeded the AlertManager will fire. 
+
+> If you want the scaling happening based on CPU and Memory usage please have a brief look at [this blog](https://stefanprodan.com/2018/kubernetes-scaleway-baremetal-arm-terraform-installer/#horizontal-pod-autoscaling)
+
+For each fired alarm the auto-scaler will add 5 instances to the affected lambda. Open FaaS has two configuration options that allow to specify the starting/minimum amount of instances and also allows to ceil the maximum amount of instances:
```
Comment (alexellis) [inline lab8.md]: Please avoid the word "lambda" as it's a loaded term - i.e. "an AWS product". Let's use the word function?

Open FaaS should be OpenFaaS.

Comment (alexellis) [reply lab8.md]: instances -> replicas

File: lab8.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+You can control the minimum amount of instances for lambda by setting `com.openfaas.scale.min`, the default value is currently `1`. 
+
+You can control the maximum amount of instances that can spawn for a lambda by setting `com.openfaas.scale.max`, the default value is currently `20`. 
+
+> Note: If you set `com.openfaas.scale.min` and `com.openfaas.scale.max` to the same value you are disabling the auto-scaling feature. So pay attention to not do this accidentally. 
+
+## Hands-on:
+
+### Preparing Prometheus:
+
```
Comment (alexellis) [inline lab8.md]: Please assume the port is 9090 and remove this section up until line 40 where you ask them to open a browser.

File: lab8.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ Let's add a graph with all successful invocation of the deployed lambdas. We can do this by executing `gateway_function_invocation_total{code="200"}` as a query. Resulting in a page that looks like this: 
+
+ ![](./screenshot/prometheus_graph.png)
+
+ Go ahead an open a new tab in which you navigate to the alert section using `http://localhost:9090/alerts`. On this page, you can later see when the threshold for the `request per seconds` is exceeded.
+
+ ![](./screenshot/prometheus_alerts.png)
+
+ ### Invoking Lambda:
```
Comment (alexellis) [inline lab8.md]: We can't use the term "Lambda" here (sorry)

File: lab8.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+ Using the following little bash-command we can constantly invoke the `func_nodeinfo` lambda.
+
+ ```bash
+$ while [ true ]; do curl -X POST http://localhost:8080/function/func_nodeinfo; done;
+ ```
+
+### Observing Auto Scaling:
```
Comment (alexellis) [inline lab8.md]: Auto-scaling

Please also mention that this is visible in the API Gateway Portal / UI and via `faas-cli list`

File: lab8.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+* At this point, it is expected you already have set up Open FaaS for Docker Swarm as described in [Lab 1](https://github.com/openfaas/workshop/blob/master/lab1.md).
+
+* As we need a way to invoke our lambda to demonstrate auto-scaling it is expected that you have access to [Curl](http://macappstore.org/curl/) or another to which allows you to do an HTTP-Request.
+
+
+## Exploring Auto-Scaling:
+
+Before dive into a hands-on session, we will have a brief look how we can configure the auto-scaling out of the box. 
+
+Out of the box Open FaaS is set up in a way that it will auto-scale based on the received `request per seconds`. If the defined threshold for the `request per seconds` is exceeded the AlertManager will fire.
```
Comment (alexellis) [inline lab8.md]: "Open FaaS" -> OpenFaaS

**Comment by Templum** [comment]
Addressed your feedback in 2744cdc and 92d39b2

**Comment by alexellis** [comment]
Thank you Simon. I'll merge this now.

**Comment by BurtonR** [comment]
@alexellis  I ran through this lab completely and verified everything works as expected! 1439 requests to `nodeinfo` later... :smile: 

Nice work @Templum!

**Comment by alexellis** [comment]
1439 is that all? :smiley_cat:

**Pull request merged**